pkg/controller/download_request_controller.go

```diff
@@ -214,8 +219,24 @@ const signedURLTTL = 10 * time.Minute
 func (c *downloadRequestController) generatePreSignedURL(downloadRequest *v1.DownloadRequest) error {
 	update := downloadRequest.DeepCopy()
 
-	var err error
-	update.Status.DownloadURL, err = c.backupService.CreateSignedURL(downloadRequest.Spec.Target, c.bucket, signedURLTTL)
+	var (
+		directory string
+		err       error
+	)
+
+	switch downloadRequest.Spec.Target.Kind {
+	case v1.DownloadTargetKindRestoreLog, v1.DownloadTargetKindRestoreResults:
+		restore, err := c.restoreLister.Restores(v1.DefaultNamespace).Get(downloadRequest.Spec.Target.Name)
+		if err != nil {
+			return errors.Wrap(err, "error getting Restore")
+		}
+
+		directory = restore.Spec.BackupName
+	default:
+		directory = downloadRequest.Spec.Target.Name
```

Comment:
This flow technically isn't used anywhere - do you think it's worth it?

Reply:
The `default` case? That should catch anything backup-related (logs, tarball)